Homework4: Anton Stychnevsky

Stychnevsky opened the pull request:
Homework # 4

Stychnevsky committed:
Homework # 4

Stychnevsky committed:
fixed import according PEP8

Stychnevsky committed:
fixed import according PEP8 (2)

imports now in alphabetical order

Stychnevsky commented on HW4 - task1.py:
```diff
@@ -0,0 +1,22 @@
+        calls_history.append({'func': func.__name__,
+                              'end_time': strftime("%Y-%m-%d %T", gmtime()),
+                              'result': func_call_result
+                              })
```
There is frequent situation when we have some function\input which return values(more than one and similar in every call), and this function\input is called unknown number of times (like "calls_history" in my program). What is the best type to save such sequences? I used list of dicts, but may be list of lists or list of tuples is better? Or may be it's better to create a class?

belovmd replied:
you solution is OK

Stychnevsky committed:
add second version of decorator  in task1

Stychnevsky committed:
fixed according PEP8

belovmd commented on HW4 - task2.py:
```diff
@@ -0,0 +1,21 @@
+    if len(lst) == 1:
+        return str(lst[0])
+
+    new_lst = ''
```
why variable with name list keeps string?

belovmd commented on HW4 - task1.py:
```diff
@@ -0,0 +1,48 @@
+    There is decision for first version of task - decorator save all functions
+    called during program running
+    """
+    if not hasattr(func_calls_history_v1, 'calls_history'):
```
good solution

belovmd reviewed (commented):
please address these minor issues

Stychnevsky committed:
refactored task2

- add parenthesis to negative values
  example: '-5-(-3),0,4-7'
- add 3 variables (prev, curr, nxt), which make program longer but easier to read
- rename variable "new_lst" to "short_form"

belovmd reviewed (commented):
9
